### PR TITLE
Add ASCII mode

### LINKED ISSUE #58: Ascii mode
## Problem

Old school ascii mode, just like in gnuchess. The player has uppercase letters and the enemy has lowercase letters.
p = pawn, k = king, q = queen, b = bishop, r = rook, n = knight

image

### PR DESCRIPTION
**Pull request by pSnehanshu**
# Add ASCII mode

## Description

Fixes #58 

* A new option "Display mode" is added, selecting which will toggle the mode between "Default" and "ASCII".
* Using the GNU Chess notation for the pieces (e.g. R-Rook, N-Knight, K-King, etc.)
* Upper-case for the current player, and lower-case for the opponent.
* I haven't added the `--ascii` flag (suggested [here](https://github.com/thomas-mauran/chess-tui/issues/58#issuecomment-2014632510)). Please let me know if this flag will be desirable.

image

image

## How Has This Been Tested?

I have tested this manually, and all the existing test cases pass. However, I haven't written any tests for board rendering, since there doesn't seem to be any such existing tests.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by thomas-mauran — CHANGES_REQUESTED** [review]
Nice code thanks for the contribution, while reading I noted a few ideas that might be cool to also implement in this pr: 
- as mentioned in the code this PR could be a good start for having a more general theme selector in the game. The ascii theme could be different in style and color while adding as you did the ascii chess characters. Maybe the board could only use black and white and no background for the cell but only a border when the cell is white ? I don't know at all what could be done here but you can be creative any ideas are welcomed !
- Can you center vertically the letters so they are in the middle of the cell ?
- I was having trouble playing this the GNU Chess notation because the letter color changes depending of the cell color, with a more black and white theme with a black background and different types of cell borders this might be easier to play, like the screenshot mentioned in the issue #58 where the background stays black and the pieces are easier to differentiate due to their size
- run cargo clippy 

Apart from those propositions thanks again for your contribution !

**Comment by pSnehanshu** [comment]
Output of `cargo clippy`

```bash
    Checking chess-tui v1.2.0 (/home/snehanshu/projects/chess-tui)
warning: equality checks against false can be replaced by a negation
   --> src/board.rs:318:28
    |
318 |                         if self.is_promotion == false {
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try simplifying it as shown: `!self.is_promotion`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#bool_comparison
    = note: `#[warn(clippy::bool_comparison)]` on by default

warning: this match could be written as a `let` statement
   --> src/board.rs:366:17
    |
366 | /                 match (
367 | |                     get_piece_type(self.board, [i, j]),
368 | |                     get_piece_color(self.board, [i, j]),
369 | |                 ) {
...   |
397 | |                     }
398 | |                 }
    | |_________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#match_single_binding
    = note: `#[warn(clippy::match_single_binding)]` on by default
help: consider using a `let` statement
    |
366 ~                 let (piece_type, piece_color) = (
367 +                     get_piece_type(self.board, [i, j]),
368 +                     get_piece_color(self.board, [i, j]),
369 +                 );
370 +                 match PieceType::piece_to_fen_enum(piece_type, piece_color) {
371 +                     // Pattern match directly on the result of piece_to_fen_enum
372 +                     "" => {
373 +                         // Check if the string is not empty before using chars().last()
374 +                         if let Some(last_char) = result.chars().last() {
375 +                             if last_char.is_ascii_digit() {
376 +                                 let incremented_char = char::from_digit(
377 +                                     last_char.to_digit(10).unwrap_or(0) + 1,
378 +                                     10,
379 +                                 )
380 +                                 .unwrap_or_default();
381 +                                 // Remove the old number and add the new incremented one
382 +                                 result.pop();
383 +                                 result.push_str(incremented_char.to_string().as_str());
384 +                             } else {
385 +                                 result.push('1');
386 +                             }
387 +                         } else {
388 +                             result.push('1');
389 +                         }
390 +                     }
391 +                     letter => {
392 +                         // If the result is not an empty string, push '1'
393 +                         result.push_str(letter);
394 +                     }
395 +                 }
    |

warning: calling `push_str()` using a single-character string literal
   --> src/board.rs:419:17
    |
419 |                 result.push_str("q");
    |                 ^^^^^^^^^^^^^^^^^^^^ help: consider using `push` with a character literal: `result.push('q')`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_char_add_str
    = note: `#[warn(clippy::single_char_add_str)]` on by default

warning: calling `push_str()` using a single-character string literal
   --> src/board.rs:441:21
    |
441 |                     result.push_str(" ");
    |                     ^^^^^^^^^^^^^^^^^^^^ help: consider using `push` with a character literal: `result.push(' ')`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_char_add_str

warning: calling `push_str()` using a single-character string literal
   --> src/board.rs:449:9
    |
449 |         result.push_str(" ");
    |         ^^^^^^^^^^^^^^^^^^^^ help: consider using `push` with a character literal: `result.push(' ')`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_char_add_str

warning: calling `push_str()` using a single-character string literal
   --> src/board.rs:452:9
    |
452 |         result.push_str(" ");
    |         ^^^^^^^^^^^^^^^^^^^^ help: consider using `push` with a character literal: `result.push(' ')`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_char_add_str

warning: unneeded `return` statement
   --> src/board.rs:470:17
    |
470 |                 return false;
    |                 ^^^^^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_return
    = note: `#[warn(clippy::needless_return)]` on by default
help: remove `return`
    |
470 -                 return false;
470 +                 false
    |

warning: `if` chain can be rewritten with `match`
   --> src/board.rs:552:13
    |
552 | /             if distance > 0 {
553 | |                 row_index_rook = 3;
554 | |                 if self.is_game_against_bot && self.player_turn == PieceColor::Black {
555 | |                     to_x = 0;
...   |
561 | |                 }
562 | |             }
    | |_____________^
    |
    = help: consider rewriting the `if` chain to use `cmp` and `match`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#comparison_chain
    = note: `#[warn(clippy::comparison_chain)]` on by default

warning: `chess-tui` (lib) generated 8 warnings (run `cargo clippy --fix --lib -p chess-tui` to apply 6 suggestions)
    Finished dev [unoptimized + debuginfo] target(s) in 0.69s
snehanshu@SM-PF3WY6M6:~/projects/chess-tui$ cargo clippy > clippy
warning: equality checks against false can be replaced by a negation
   --> src/board.rs:318:28
    |
318 |                         if self.is_promotion == false {
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try simplifying it as shown: `!self.is_promotion`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#bool_comparison
    = note: `#[warn(clippy::bool_comparison)]` on by default

warning: this match could be written as a `let` statement
   --> src/board.rs:366:17
    |
366 | /                 match (
367 | |                     get_piece_type(self.board, [i, j]),
368 | |                     get_piece_color(self.board, [i, j]),
369 | |                 ) {
...   |
397 | |                     }
398 | |                 }
    | |_________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#match_single_binding
    = note: `#[warn(clippy::match_single_binding)]` on by default
help: consider using a `let` statement
    |
366 ~                 let (piece_type, piece_color) = (
367 +                     get_piece_type(self.board, [i, j]),
368 +                     get_piece_color(self.board, [i, j]),
369 +                 );
370 +                 match PieceType::piece_to_fen_enum(piece_type, piece_color) {
371 +                     // Pattern match directly on the result of piece_to_fen_enum
372 +                     "" => {
373 +                         // Check if the string is not empty before using chars().last()
374 +                         if let Some(last_char) = result.chars().last() {
375 +                             if last_char.is_ascii_digit() {
376 +                                 let incremented_char = char::from_digit(
377 +                                     last_char.to_digit(10).unwrap_or(0) + 1,
378 +                                     10,
379 +                                 )
380 +                                 .unwrap_or_default();
381 +                                 // Remove the old number and add the new incremented one
382 +                                 result.pop();
383 +                                 result.push_str(incremented_char.to_string().as_str());
384 +                             } else {
385 +                                 result.push('1');
386 +                             }
387 +                         } else {
388 +                             result.push('1');
389 +                         }
390 +                     }
391 +                     letter => {
392 +                         // If the result is not an empty string, push '1'
393 +                         result.push_str(letter);
394 +                     }
395 +                 }
    |

warning: calling `push_str()` using a single-character string literal
   --> src/board.rs:419:17
    |
419 |                 result.push_str("q");
    |                 ^^^^^^^^^^^^^^^^^^^^ help: consider using `push` with a character literal: `result.push('q')`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_char_add_str
    = note: `#[warn(clippy::single_char_add_str)]` on by default

warning: calling `push_str()` using a single-character string literal
   --> src/board.rs:441:21
    |
441 |                     result.push_str(" ");
    |                     ^^^^^^^^^^^^^^^^^^^^ help: consider using `push` with a character literal: `result.push(' ')`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_char_add_str

warning: calling `push_str()` using a single-character string literal
   --> src/board.rs:449:9
    |
449 |         result.push_str(" ");
    |         ^^^^^^^^^^^^^^^^^^^^ help: consider using `push` with a character literal: `result.push(' ')`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_char_add_str

warning: calling `push_str()` using a single-character string literal
   --> src/board.rs:452:9
    |
452 |         result.push_str(" ");
    |         ^^^^^^^^^^^^^^^^^^^^ help: consider using `push` with a character literal: `result.push(' ')`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_char_add_str

warning: unneeded `return` statement
   --> src/board.rs:470:17
    |
470 |                 return false;
    |                 ^^^^^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_return
    = note: `#[warn(clippy::needless_return)]` on by default
help: remove `return`
    |
470 -                 return false;
470 +                 false
    |

warning: `if` chain can be rewritten with `match`
   --> src/board.rs:552:13
    |
552 | /             if distance > 0 {
553 | |                 row_index_rook = 3;
554 | |                 if self.is_game_against_bot && self.player_turn == PieceColor::Black {
555 | |                     to_x = 0;
...   |
561 | |                 }
562 | |             }
    | |_____________^
    |
    = help: consider rewriting the `if` chain to use `cmp` and `match`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#comparison_chain
    = note: `#[warn(clippy::comparison_chain)]` on by default

warning: `chess-tui` (lib) generated 8 warnings (run `cargo clippy --fix --lib -p chess-tui` to apply 6 suggestions)
    Finished dev [unoptimized + debuginfo] target(s) in 0.02s
```

**Comment by pSnehanshu** [comment]
How does this look?

image

File: src/board.rs [hunk 1]
```diff
@@ -776,15 +784,33 @@ impl Board {
...
+                            // pieces belonging to the player on bottom will be upper case
+                            Some(PieceColor::White) => piece_enum.to_uppercase(),
+                            // Pass through original value
+                            None => String::from(piece_enum),
+                        };
+
+                        // Place the pieces on the board
+                        Paragraph::new(piece_enum_case)
+                    }
+                };
```
[the remark this answers is not in the document]
Comment (pSnehanshu) [reply src/board.rs]: Where do you think is the best place to move this code to?

**Comment by thomas-mauran** [comment]
> How does this look?
> 
> image

I like a lot this version, I just saw that ratatui got a [underline](https://docs.rs/ratatui/latest/ratatui/style/trait.Stylize.html#method.underlined) and [bold](https://docs.rs/ratatui/latest/ratatui/style/trait.Stylize.html#method.bold) style option for text can we try that for the elements of the current player to make it even easier to see on the board

**Comment by thomas-mauran** [comment]
> Output of `cargo clippy`
> 
> ```shell
>     Checking chess-tui v1.2.0 (/home/snehanshu/projects/chess-tui)
> warning: equality checks against false can be replaced by a negation
>    --> src/board.rs:318:28
>     |
> 318 |                         if self.is_promotion == false {
>     |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try simplifying it as shown: `!self.is_promotion`
>     |
>     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#bool_comparison
>     = note: `#[warn(clippy::bool_comparison)]` on by default
> 
> warning: this match could be written as a `let` statement
>    --> src/board.rs:366:17
>     |
> 366 | /                 match (
> 367 | |                     get_piece_type(self.board, [i, j]),
> 368 | |                     get_piece_color(self.board, [i, j]),
> 369 | |                 ) {
> ...   |
> 397 | |                     }
> 398 | |                 }
>     | |_________________^
>     |
>     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#match_single_binding
>     = note: `#[warn(clippy::match_single_binding)]` on by default
> help: consider using a `let` statement
>     |
> 366 ~                 let (piece_type, piece_color) = (
> 367 +                     get_piece_type(self.board, [i, j]),
> 368 +                     get_piece_color(self.board, [i, j]),
> 369 +                 );
> 370 +                 match PieceType::piece_to_fen_enum(piece_type, piece_color) {
> 371 +                     // Pattern match directly on the result of piece_to_fen_enum
> 372 +                     "" => {
> 373 +                         // Check if the string is not empty before using chars().last()
> 374 +                         if let Some(last_char) = result.chars().last() {
> 375 +                             if last_char.is_ascii_digit() {
> 376 +                                 let incremented_char = char::from_digit(
> 377 +                                     last_char.to_digit(10).unwrap_or(0) + 1,
> 378 +                                     10,
> 379 +                                 )
> 380 +                                 .unwrap_or_default();
> 381 +                                 // Remove the old number and add the new incremented one
> 382 +                                 result.pop();
> 383 +                                 result.push_str(incremented_char.to_string().as_str());
> 384 +                             } else {
> 385 +                                 result.push('1');
> 386 +                             }
> 387 +                         } else {
> 388 +                             result.push('1');
> 389 +                         }
> 390 +                     }
> 391 +                     letter => {
> 392 +                         // If the result is not an empty string, push '1'
> 393 +                         result.push_str(letter);
> 394 +                     }
> 395 +                 }
>     |
> 
> warning: calling `push_str()` using a single-character string literal
>    --> src/board.rs:419:17
>     |
> 419 |                 result.push_str("q");
>     |                 ^^^^^^^^^^^^^^^^^^^^ help: consider using `push` with a character literal: `result.push('q')`
>     |
>     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_char_add_str
>     = note: `#[warn(clippy::single_char_add_str)]` on by default
> 
> warning: calling `push_str()` using a single-character string literal
>    --> src/board.rs:441:21
>     |
> 441 |                     result.push_str(" ");
>     |                     ^^^^^^^^^^^^^^^^^^^^ help: consider using `push` with a character literal: `result.push(' ')`
>     |
>     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_char_add_str
> 
> warning: calling `push_str()` using a single-character string literal
>    --> src/board.rs:449:9
>     |
> 449 |         result.push_str(" ");
>     |         ^^^^^^^^^^^^^^^^^^^^ help: consider using `push` with a character literal: `result.push(' ')`
>     |
>     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_char_add_str
> 
> warning: calling `push_str()` using a single-character string literal
>    --> src/board.rs:452:9
>     |
> 452 |         result.push_str(" ");
>     |         ^^^^^^^^^^^^^^^^^^^^ help: consider using `push` with a character literal: `result.push(' ')`
>     |
>     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_char_add_str
> 
> warning: unneeded `return` statement
>    --> src/board.rs:470:17
>     |
> 470 |                 return false;
>     |                 ^^^^^^^^^^^^
>     |
>     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_return
>     = note: `#[warn(clippy::needless_return)]` on by default
> help: remove `return`
>     |
> 470 -                 return false;
> 470 +                 false
>     |
> 
> warning: `if` chain can be rewritten with `match`
>    --> src/board.rs:552:13
>     |
> 552 | /             if distance > 0 {
> 553 | |                 row_index_rook = 3;
> 554 | |                 if self.is_game_against_bot && self.player_turn == PieceColor::Black {
> 555 | |                     to_x = 0;
> ...   |
> 561 | |                 }
> 562 | |             }
>     | |_____________^
>     |
>     = help: consider rewriting the `if` chain to use `cmp` and `match`
>     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#comparison_chain
>     = note: `#[warn(clippy::comparison_chain)]` on by default
> 
> warning: `chess-tui` (lib) generated 8 warnings (run `cargo clippy --fix --lib -p chess-tui` to apply 6 suggestions)
>     Finished dev [unoptimized + debuginfo] target(s) in 0.69s
> snehanshu@SM-PF3WY6M6:~/projects/chess-tui$ cargo clippy > clippy
> warning: equality checks against false can be replaced by a negation
>    --> src/board.rs:318:28
>     |
> 318 |                         if self.is_promotion == false {
>     |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try simplifying it as shown: `!self.is_promotion`
>     |
>     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#bool_comparison
>     = note: `#[warn(clippy::bool_comparison)]` on by default
> 
> warning: this match could be written as a `let` statement
>    --> src/board.rs:366:17
>     |
> 366 | /                 match (
> 367 | |                     get_piece_type(self.board, [i, j]),
> 368 | |                     get_piece_color(self.board, [i, j]),
> 369 | |                 ) {
> ...   |
> 397 | |                     }
> 398 | |                 }
>     | |_________________^
>     |
>     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#match_single_binding
>     = note: `#[warn(clippy::match_single_binding)]` on by default
> help: consider using a `let` statement
>     |
> 366 ~                 let (piece_type, piece_color) = (
> 367 +                     get_piece_type(self.board, [i, j]),
> 368 +                     get_piece_color(self.board, [i, j]),
> 369 +                 );
> 370 +                 match PieceType::piece_to_fen_enum(piece_type, piece_color) {
> 371 +                     // Pattern match directly on the result of piece_to_fen_enum
> 372 +                     "" => {
> 373 +                         // Check if the string is not empty before using chars().last()
> 374 +                         if let Some(last_char) = result.chars().last() {
> 375 +                             if last_char.is_ascii_digit() {
> 376 +                                 let incremented_char = char::from_digit(
> 377 +                                     last_char.to_digit(10).unwrap_or(0) + 1,
> 378 +                                     10,
> 379 +                                 )
> 380 +                                 .unwrap_or_default();
> 381 +                                 // Remove the old number and add the new incremented one
> 382 +                                 result.pop();
> 383 +                                 result.push_str(incremented_char.to_string().as_str());
> 384 +                             } else {
> 385 +                                 result.push('1');
> 386 +                             }
> 387 +                         } else {
> 388 +                             result.push('1');
> 389 +                         }
> 390 +                     }
> 391 +                     letter => {
> 392 +                         // If the result is not an empty string, push '1'
> 393 +                         result.push_str(letter);
> 394 +                     }
> 395 +                 }
>     |
> 
> warning: calling `push_str()` using a single-character string literal
>    --> src/board.rs:419:17
>     |
> 419 |                 result.push_str("q");
>     |                 ^^^^^^^^^^^^^^^^^^^^ help: consider using `push` with a character literal: `result.push('q')`
>     |
>     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_char_add_str
>     = note: `#[warn(clippy::single_char_add_str)]` on by default
> 
> warning: calling `push_str()` using a single-character string literal
>    --> src/board.rs:441:21
>     |
> 441 |                     result.push_str(" ");
>     |                     ^^^^^^^^^^^^^^^^^^^^ help: consider using `push` with a character literal: `result.push(' ')`
>     |
>     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_char_add_str
> 
> warning: calling `push_str()` using a single-character string literal
>    --> src/board.rs:449:9
>     |
> 449 |         result.push_str(" ");
>     |         ^^^^^^^^^^^^^^^^^^^^ help: consider using `push` with a character literal: `result.push(' ')`
>     |
>     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_char_add_str
> 
> warning: calling `push_str()` using a single-character string literal
>    --> src/board.rs:452:9
>     |
> 452 |         result.push_str(" ");
>     |         ^^^^^^^^^^^^^^^^^^^^ help: consider using `push` with a character literal: `result.push(' ')`
>     |
>     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_char_add_str
> 
> warning: unneeded `return` statement
>    --> src/board.rs:470:17
>     |
> 470 |                 return false;
>     |                 ^^^^^^^^^^^^
>     |
>     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_return
>     = note: `#[warn(clippy::needless_return)]` on by default
> help: remove `return`
>     |
> 470 -                 return false;
> 470 +                 false
>     |
> 
> warning: `if` chain can be rewritten with `match`
>    --> src/board.rs:552:13
>     |
> 552 | /             if distance > 0 {
> 553 | |                 row_index_rook = 3;
> 554 | |                 if self.is_game_against_bot && self.player_turn == PieceColor::Black {
> 555 | |                     to_x = 0;
> ...   |
> 561 | |                 }
> 562 | |             }
>     | |_____________^
>     |
>     = help: consider rewriting the `if` chain to use `cmp` and `match`
>     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#comparison_chain
>     = note: `#[warn(clippy::comparison_chain)]` on by default
> 
> warning: `chess-tui` (lib) generated 8 warnings (run `cargo clippy --fix --lib -p chess-tui` to apply 6 suggestions)
>     Finished dev [unoptimized + debuginfo] target(s) in 0.02s
> ```

to fix the lint with clippy I often use this command: `cargo clippy --fix --allow-dirty --allow-staged` you can try running that if you can

**Comment by pSnehanshu** [comment]
> to fix the lint with clippy I often use this command: cargo clippy --fix --allow-dirty --allow-staged you can try running that if you can

@thomas-mauran sure, but those warnings aren't related to my changes, so I don't think that's relevant in this PR.

**Comment by thomas-mauran** [comment]
> > to fix the lint with clippy I often use this command: cargo clippy --fix --allow-dirty --allow-staged you can try running that if you can
> 
> @thomas-mauran sure, but those warnings aren't related to my changes, so I don't think that's relevant in this PR.

ah yes my bad, I will fix them in another small lint pr when adding the lint test to the ci 👍

**Comment by pSnehanshu** [comment]
image

I underlined the current user's pieces. I didn't bold anything because I think it looks fine without it.

**Comment by thomas-mauran** [comment]
> image
> 
> I underlined the current user's pieces. I didn't bold anything because I think it looks fine without it.

looking great like that mate !

File: src/board.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (thomas-mauran) [reply src/board.rs]: Maybe we could put that in utils ? I don't know if you have a better idea

[the remark this answers is not in the document]
Comment (pSnehanshu) [reply src/board.rs]: Yeah, that makes sense. I moved it, let me know if this looks good.

**Pull request merged**